### ISSUE TITLE
Cannot close dropdowns in safari 

### ISSUE THREAD

**Issue by aaronmaturen** [cengage/react-magma#340]
When clicking on the DropdownButton to close the DropDown in safari it (very) briefly closes and then reopens

https://user-images.githubusercontent.com/507070/128946780-16edb90b-9f77-4d4b-9bc2-24bc98d8135b.mp4

**Comment by aaronmaturen** [comment]
After looking at this for awhile, it looks like since we do `toggleRef.current.focus()` to set the focus styles for the dropdown button safari will call `onBlur` before `onClick` even when the currently focused element is losing focus to itself. All the call backs either have `body` or `null` for the next element which makes it hard to find something that says it is a self click...

**Comment by aaronmaturen** [comment]
One possibility is to add `[aria-expanded="true"]` to the `StyledButton:focus` styles so we don't need to set the focus manually...

**Comment by aaronmaturen** [comment]
Also... it appears that we can't tab into a dropdown menu by default... that seems unintentional...

**Comment by orion-cengage** [comment]
I definitely don't like the idea of the focus outline being around both the button and the currently focused menu item.
